src/java/com/reucon/openfire/plugin/archive/impl/JdbcPersistenceManager.java

```diff
@@ -489,22 +489,31 @@ private int bindConversationParameters(Date startDate, Date endDate, String owne
             if (xmppResultSet.getIndex() != null) {
                 firstIndex = xmppResultSet.getIndex();
             } else if (xmppResultSet.getAfter() != null) {
-                final Long needle;
+                Long needle;
                 if ( useStableID ) {
                     needle = ConversationManager.getMessageIdForStableId( new JID(ownerJid), xmppResultSet.getAfter() );
+                    if (needle==null)
+                    	needle = new Long((new java.util.Date()).getTime());
```

Comment:
Can you please use a consistent amount of indentation, and always use brackets (even if there's just one line in the if-statement)